File: drebin_store/Services/Models/Order.cs
using System;

namespace drebin_store.Services.Models
{
    public class Order
    {
        public int Id { get; set; }

        public User User { get; set; }

        public Product Product { get; set; }

        public bool IsCompleted { get; set; }

        public DateTime OrderTimeStamp { get; set; }

        public DateTime CompletionTimeStamp { get; set; }
    }
}

File: drebin_store/ClientApp/src/app/administration/orders-manager/models/order.ts
import { User } from 'src/app/models/user';
import { Product } from 'src/app/store/models/product';
import { OrderState } from './order-state';

export class Order {
    id: number;
    user: User;
    product: Product;
    orderState: OrderState;
    orderTimeStamp: Date;
    completionTimeStamp: Date;

    constructor(order?: Order) {
        if (order != null) {
            this.id = order.id;
            this.user = order.user;
            this.product = order.product;
            this.orderState = order.orderState;
            this.orderTimeStamp = order.orderTimeStamp;
            this.completionTimeStamp = order.completionTimeStamp;
        }
    }
}
